### PR TITLE
feat: add pagination component to components directory

### LINKED ISSUE #3266: Create a pagination component
## User Story
As a user, I would like to be able to paginate through large data sets in the web UI using an intuitive pagination widget.

## Details
We will need this component for the new Audit page
Screen Shot 2022-07-28 at 3 33 17 PM

## AC

- [ ] component looks like the mock provided [here](https://www.figma.com/file/oy7echtkBVRuOCMBVSrr4T/Auditing?node-id=0%3A1)
- [ ] component is accessible via keyboard
- [ ] component is dark-themed
- [ ] component implementation is based on our [pagination struct](https://github.com/coder/coder/blob/main/codersdk/pagination.go#L11)
- [ ] component is generic enough to be used in many places
- [ ] component has reasonable states for no pages returned
- [ ] component has reasonable state for 1 page returned
- [ ] component has reasonable state for n pages returned
- [ ] component is flexible and gives a lot of responsibility to the parent

## Definition of Done

- [ ] Above AC is met
- [ ] New component is created and lives in our `/components` directory
- [ ] component has tests and stories

### PR DESCRIPTION
**Pull request by Kira-Pilot**
Resolves #3266 

Here's an accessible pagination widget for the upcoming audit work (and for future pagination needs).
Kapture 2022-07-29 at 10 43 51

## States

**Unknown number of pages:**
Screen Shot 2022-07-29 at 12 37 58 PM

**Less than 8 pages:**
Screen Shot 2022-07-29 at 12 38 07 PM

**Many pages, user at the start:**
Screen Shot 2022-07-29 at 12 39 19 PM

**Many pages, user in the middle:**
Screen Shot 2022-07-29 at 12 40 32 PM

**Many pages, user at the end:**
Screen Shot 2022-07-29 at 12 41 32 PM

### CODE REVIEW

File: site/src/components/PaginationWidget/PaginationWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  activePage = 1,
+  containerStyle,
+}: PaginationWidgetProps): JSX.Element | null => {
+  const numPages = numRecords ? Math.ceil(numRecords / numRecordsPerPage) : 0
+  const firstPageActive = activePage === 1 && numPages !== 0
+  const lastPageActive = activePage === numPages && numPages !== 0
+
+  const styles = useStyles()
+
+  const buildPagedList = () => {
```
Comment (presleyp) [inline site/src/components/PaginationWidget/PaginationWidget.tsx:54]: I like to pull pure functions out of components as much as possible, maybe that can be done here? Then they can be unit tested easily, although I know you've tested the component, so it's up to you.

Comment (Kira-Pilot) [reply site/src/components/PaginationWidget/PaginationWidget.tsx:54]: Sure, I can take a stab at that!

Comment (BrunoQuaresma) [reply site/src/components/PaginationWidget/PaginationWidget.tsx:54]: I have a diff opinion, I think it is easier to manage/find code by keeping them close to where they are used, and it looks like this function only will be used on this component so I would keep it in the component file but it is up to you.

**Review by BrunoQuaresma — APPROVED** [review]
Nice work!

File: site/src/components/PaginationWidget/PaginationWidget.tsx (hunk 1 above)
Comment (Kira-Pilot) [reply site/src/components/PaginationWidget/PaginationWidget.tsx]: @presleyp @BrunoQuaresma I compromised here and added some unit tests but didn't separate the function from its component file so it should still be easy to find.

Comment (presleyp) [reply site/src/components/PaginationWidget/PaginationWidget.tsx]: Great! I think separating the function from the component is the important part.

**Pull request merged**